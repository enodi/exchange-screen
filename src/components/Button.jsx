import React from "react";

const Button = props => (
  <button className={`button ${props.color} ${props.size}`}>
    {props.text}
  </button>
);

export default Button;
