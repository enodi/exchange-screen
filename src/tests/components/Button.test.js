import React from "react";
import { shallow } from "enzyme";
import Button from "components/Button";

describe("Button Component", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<Button text="Exchange" />);
  });

  it("should render Button component correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("renders the Button text", () => {
    expect(wrapper.find("button").text()).toEqual("Exchange");
  });
});
