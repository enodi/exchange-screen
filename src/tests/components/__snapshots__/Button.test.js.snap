// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component should render Button component correctly 1`] = `
<button
  className="button undefined undefined"
>
  Exchange
</button>
`;
